lib/ssf/local_buffering_client.rb

```diff
@@ -0,0 +1,14 @@
+module SSF
+  class LocalBufferingClient < Client
+
+    attr_reader :buffer
+
+    def initialize()
+      @buffer = []
+    end
+
+    def send_to_socket(message)
+      @buffer << message
```

Comment:
Yup! That's the caller's problem. We could add a clear method for convenience though!